tests/integration/test_osm_mysql.py

```diff
@@ -0,0 +1,166 @@
+APP_NAME = METADATA["name"]
+CLUSTER_NAME = "test_cluster"
+
+
+# TODO: deploy and relate osm-grafana once it can be use with MySQL Group Replication
+@pytest.mark.order(1)
+@pytest.mark.osm_mysql_tests
+async def test_deploy_and_relate_osm_bundle(ops_test: OpsTest) -> None:
+    """Test the deployment and relation with osm bundle with mysql replacing mariadb."""
+    async with ops_test.fast_forward():
+        charm = await ops_test.build_charm(".")
+        resources = {"mysql-image": METADATA["resources"]["mysql-image"]["upstream-source"]}
+        config = {
+            "osm-mysql-interface-user": "keystone",
+            "osm-mysql-interface-database": "keystone",
+        }
+
+        osm_pol_resources = {
+            "image": "opensourcemano/pol:testing-daily",
+        }
+
+        await asyncio.gather(
+            ops_test.model.deploy(
+                charm, application_name=APP_NAME, resources=resources, config=config, num_units=3
+            ),
+            # Deploy the osm-keystone charm
+            # (using ops_test.juju instead of ops_test.deploy as the latter does
+            # not correctly deploy with the correct resources)
+            ops_test.juju(
+                "deploy",
+                "--channel=latest/edge",
+                "--trust",
+                "--resource",
+                "keystone-image=opensourcemano/keystone:testing-daily",
+                "osm-keystone",
+                "osm-keystone",
+            ),
+            ops_test.model.deploy(
+                "osm-pol",
+                application_name="osm-pol",
+                channel="latest/candidate",
+                resources=osm_pol_resources,
+            ),
+            ops_test.model.deploy("charmed-osm-kafka-k8s", application_name="osm-kafka"),
+            ops_test.model.deploy("charmed-osm-zookeeper-k8s", application_name="osm-zookeeper"),
+            ops_test.model.deploy("charmed-osm-mongodb-k8s", application_name="osm-mongodb"),
+        )
+
+        # cannot block until "osm-keystone" units are available since they are not
+        # registered with ops_test.model.applications (due to the way it's deployed)
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications[APP_NAME].units) == 3
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-pol"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-kafka"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-zookeeper"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-mongodb"].units) == 1
+        )
+
+        await ops_test.model.relate("osm-kafka:zookeeper", "osm-zookeeper:zookeeper")
+        await ops_test.model.block_until(
```

Comment:
Consider adding `timeout`'s to all `block_until` calls.

Reply:
added in [750358c](https://github.com/canonical/mysql-k8s-operator/pull/88/commits/750358c2a892641c312e45f948db18bc9efe8924)